Refactor /api/v1/get_authenticated_users

Kartones opened the pull request:
Closes #3710

Cartofante commented:
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3864/))

Kartones commented:
@juanignaciosl CR please.

Just moved to `Carto::Api` as it is public dashboard code, checked it works and removed "can fork" functionality as it is no longer needed now uses "one-click" import

juanignaciosl commented on app/controllers/admin/visualizations_controller.rb:
```diff
@@ -262,10 +262,6 @@ def public_map
   def show_organization_public_map
     return(embed_forbidden) unless org_user_has_map_permissions?(current_user, @visualization)
 
-    @can_fork = @visualization.related_tables.map { |t|
```
Why is this not needed anymore?

Kartones replied:
Because we now use the "one click import". In fact there is no use of that variable at any template

juanignaciosl commented:
:+1: (my comment is just curiosity).
